lib/puppet/defaults.rb

```diff
@@ -32,6 +32,20 @@ def self.valid_file_checksum_types
       %w[md5 md5lite sha256 sha256lite sha384 sha512 sha224 sha1 sha1lite mtime ctime]
   end
 
+  def self.default_basemodulepath
+    if Puppet::Util::Platform.windows?
+      path = ['$codedir/modules']
+      installdir = Facter.value(:env_windows_installdir)
+      if installdir
+        path << "#{installdir}/puppet/modules"
+        path << "#{installdir}/puppet/vendor_modules"
+      end
+      path.join(File::PATH_SEPARATOR)
+    else
+      '$codedir/modules:/opt/puppetlabs/puppet/modules:/opt/puppetlabs/puppet/vendor_modules'
```

Comment:
can we not define magic strings '/opt/puppetlabs/puppet' in the middle of a method and instead declare them as constants at the top? This should be settable by the user during compilation if they so choose during compilation.